### PR TITLE
style update + moire off

### PR DESCRIPTION
**Pull request by 0ctagon**
Only one default style now.

### CODE REVIEW

**Comment by 0ctagon** [comment]
See: https://plothist.readthedocs.io/en/style_update/

**Comment by 0ctagon** [comment]
Need also to remove the moire for `1d_hist_simple`

File: docs/advanced/hep_examples.rst [hunk 1]
```diff
@@ -63,6 +63,7 @@ Then, to create stacked histogram:
 .. code-block:: python
 
     from plothist import compare_data_mc, add_luminosity
+    from plothist import set_style
```
Comment (cyrraz) [inline docs/advanced/hep_examples.rst]: Not needed

File: plothist/hep_plotters.py [hunk 2]
```diff
@@ -67,8 +61,8 @@ def compare_data_mc(
         The main axes for the histogram comparison. If fig, ax_main and ax_comparison are None, a new axes will be created. Default is None.
     ax_comparison : matplotlib.axes.Axes or None, optional
         The axes for the comparison plot. If fig, ax_main and ax_comparison are None, a new axes will be created. Default is None.
-    ratio_uncertainty : str, optional
-        How to treat the uncertainties of the histograms when comparison = "ratio" ("uncorrelated" for simple comparison, "split" for scaling and split hist_1 and hist_2 uncertainties). Default is "split".
+    **comparison_kwargs : optional
+        Arguments to be passed to plot_comparison(). By default for compare_data_mc(), x1_label="Data", x2_label="Pred.", comparison="ratio", and ratio_uncertainty="split".
```
Comment (cyrraz) [inline plothist/hep_plotters.py:65]: By default for compare_data_mc(), --> If they are not provided explicitly, the following arguments are passed by default:

File: plothist/plotters.py [hunk 3]
```diff
@@ -542,15 +537,21 @@ def plot_comparison(
                 lw=0,
             )
         ax.axhline(1, ls="--", lw=1.0, color="black")
-        ax.set_ylabel(r"$\frac{" + x1_label + "}{" + x2_label + "}$")
+        if comparison_ylabel is None:
+            ax.set_ylabel(r"$\frac{" + x1_label + "}{" + x2_label + "}$")
+        else:
+            ax.set_ylabel(comparison_ylabel)
```
Comment (cyrraz) [inline plothist/plotters.py:543]: Just a single `if comparison_ylabel is not None: ax.set_ylabel(comparison_ylabel)` at the end to overwrite

File: plothist/plotters.py [hunk 4]
```diff
@@ -542,15 +537,21 @@ def plot_comparison(
...
         ax.axhline(0, ls="--", lw=1.0, color="black")
-        ax.set_ylabel(
-            rf"$\frac{{ {x1_label} - {x2_label} }}{{ \sqrt{{\sigma^2_{{{x1_label}}} + \sigma^2_{{{x2_label}}}}} }} $"
-        )
+        if comparison_ylabel is None:
+            ax.set_ylabel(
+                rf"$\frac{{ {x1_label} - {x2_label} }}{{ \sqrt{{\sigma^2_{{{x1_label}}} + \sigma^2_{{{x2_label}}}}} }} $"
+            )
+        else:
+            ax.set_ylabel(comparison_ylabel)
```
Comment (cyrraz) [inline plothist/plotters.py:554]: Just a single `if comparison_ylabel is not None: ax.set_ylabel(comparison_ylabel)` at the end to overwrite

File: plothist/default_style.mplstyle [hunk 5]
```diff
@@ -7,11 +7,11 @@ text.color          : .1
 
 # Fonts
 
-font.size           : 20
+font.size           : 15
 xtick.labelsize     : 15
 ytick.labelsize     : 15
-axes.labelsize      : 20
-legend.fontsize     : 10
+axes.labelsize      : 17
```
Comment (cyrraz) [inline plothist/default_style.mplstyle:13]: 18

File: plothist/plotters.py [hunk 6]
```diff
@@ -410,11 +406,7 @@ def compare_two_hist(
         hist_2,
         ax_comparison,
         xlabel=xlabel,
-        x1_label=x1_label,
-        x2_label=x2_label,
-        comparison=comparison,
-        comparison_ylim=comparison_ylim,
-        ratio_uncertainty=ratio_uncertainty,
+        **comparison_kwargs,
```
Comment (cyrraz) [inline plothist/plotters.py:409]: Here, we want to use the same trick as in `compare_data_mc`

Comment (cyrraz) [reply plothist/plotters.py:409]: With a dictionary update at the beginning

File: plothist/plotters.py [hunk 7]
```diff
@@ -368,8 +364,8 @@ def compare_two_hist(
         The main axes for the histogram comparison. If fig, ax_main and ax_comparison are None, a new axes will be created. Default is None.
     ax_comparison : matplotlib.axes.Axes or None, optional
         The axes for the comparison plot. If fig, ax_main and ax_comparison are None, a new axes will be created. Default is None.
-    ratio_uncertainty : str, optional
-        How to treat the uncertainties of the histograms when comparison = "ratio" ("uncorrelated" for simple comparison, "split" for scaling and split hist_1 and hist_2 uncertainties). Default is "uncorrelated"
+    **comparison_kwargs : optional
+        Arguments to be passed to plot_comparison().
```
Comment (cyrraz) [inline plothist/plotters.py:368]: If they are not provided explicitly, the following arguments are passed by default: [...] (same trick as in `compare_data_mc()`)

File: plothist/plotters.py [hunk 8]
```diff
@@ -368,8 +364,8 @@ def compare_two_hist(
         The main axes for the histogram comparison. If fig, ax_main and ax_comparison are None, a new axes will be created. Default is None.
     ax_comparison : matplotlib.axes.Axes or None, optional
         The axes for the comparison plot. If fig, ax_main and ax_comparison are None, a new axes will be created. Default is None.
-    ratio_uncertainty : str, optional
-        How to treat the uncertainties of the histograms when comparison = "ratio" ("uncorrelated" for simple comparison, "split" for scaling and split hist_1 and hist_2 uncertainties). Default is "uncorrelated"
+    **comparison_kwargs : optional
+        Arguments to be passed to plot_comparison(), including the choice of the comparison function and the treatment of the uncertainties (see documentation of plot_comparison() for details)
```
Comment (cyrraz) [inline plothist/plotters.py:368]: Missing period at the end

File: plothist/plotters.py [hunk 9]
```diff
@@ -427,10 +421,11 @@ def plot_comparison(
     hist_1,
     hist_2,
     ax,
-    xlabel="x1",
-    x1_label="x1",
-    x2_label="x2",
+    xlabel="h1",
```
Comment (cyrraz) [inline plothist/plotters.py:424]: x1, not h1

File: plothist/plotters.py [hunk 10]
```diff
@@ -446,13 +441,15 @@ def plot_comparison(
     ax : matplotlib.axes.Axes
         The axes to plot the comparison.
     xlabel : str, optional
-        The label for the x-axis. Default is "x1".
-    x1_label : str, optional
-        The label for the first histogram. Default is "x1".
-    x2_label : str, optional
-        The label for the second histogram. Default is "x2".
+        The label for the x-axis. Default is "h1".
```
Comment (cyrraz) [inline plothist/plotters.py:444]: x1, not h1

File: plothist/plotters.py [hunk 11]
```diff
@@ -542,20 +539,22 @@ def plot_comparison(
...
-            rf"$\frac{{ {x1_label} - {x2_label} }}{{ \sqrt{{\sigma^2_{{{x1_label}}} + \sigma^2_{{{x2_label}}}}} }} $"
+            rf"$\frac{{ {h1_label} - {h2_label} }}{{ \sqrt{{\sigma^2_{{{h1_label}}} + \sigma^2_{{{h2_label}}}}} }} $"
         )
 
+
     xlim = (hist_1.axes[0].edges[0], hist_1.axes[0].edges[-1])
     ax.set_xlim(xlim)
     ax.set_ylim(comparison_ylim)
     ax.set_xlabel(xlabel)
+    if comparison_ylabel is not None: ax.set_ylabel(comparison_ylabel)
```
Comment (cyrraz) [inline plothist/plotters.py:557]: black would not allow that single line condition

Comment (cyrraz) [reply plothist/plotters.py:557]: or?

Comment (0ctagon) [reply plothist/plotters.py:557]: Bien vu!

**Review by cyrraz — APPROVED** [review]

**Pull request merged**